cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_auto_scaling_group.go

```diff
@@ -94,26 +94,25 @@ func (asg *AutoScalingGroup) IncreaseSize(delta int) error {
 func (asg *AutoScalingGroup) DeleteNodes(nodes []*apiv1.Node) error {
 	instances, err := asg.cloudServiceManager.GetInstances(asg.groupID)
 	if err != nil {
-		klog.Warningf("failed to get nodes from group: %s, error: %v", asg.groupID, err)
+		klog.Warningf("failed to get instances from group: %s, error: %v", asg.groupID, err)
 		return err
 	}
 
-	// TODO(RainbowMango): Check if all node in this group.
-	//  If one of the node is not belong to this group, just return error.
-
-	servers := make([]string, 0, len(instances))
+	instanceIds := make([]string, 0, len(instances))
 	for _, instance := range instances {
-		servers = append(servers, instance.Id)
+		for _, n := range nodes {
+			if n.Spec.ProviderID == instance.Id {
+				instanceIds = append(instanceIds, instance.Id)
+			}
+		}
 	}
 
-	err = asg.cloudServiceManager.DeleteServers(servers)
+	err = asg.cloudServiceManager.ScaleDownCluster(asg.groupID, instanceIds)
 	if err != nil {
-		klog.Warningf("failed to delete nodes. error: %v", err)
+		klog.Warningf("failed to scale down cluster. error: %v", err)
 		return err
 	}
 
```

Comment:
Don't we have to wait for the node group size update?

Reply:
No, after calling the api, the node group size (I'm assuming the group size is the desire instance number, we can discuss this later) will change first, then it's going to remove those instances.